hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -1138,8 +1138,12 @@ private void handleUnderReplicatedHealthy(final ContainerInfo container,
           container.containerID());
     }
 
+    State matchingReplicaState = State.CLOSED;
+    if (container.getState() == LifeCycleState.QUASI_CLOSED) {
+      matchingReplicaState = State.QUASI_CLOSED;
```

Comment:
If we are in the container=QUASI_CLOSED branch, then do we need to ensure we only replicate the largest sequenceID quasi_closed replica? There isn't any logic in `getReplicationSources` to do that right now.

Reply:
This is not required. I added a comment to this part of the code to explain why, and added a unit test to verify this behavior. Let me know if the comment makes sense.